src/plugins/topic_viewer/TopicViewer_TEST.cc

```diff
@@ -0,0 +1,146 @@
+#include <gtest/gtest.h>
+
+#include <ignition/common/Console.hh>
+
+#include "test_config.h"  // NOLINT(build/include)
+#include "ignition/gui/Application.hh"
+#include "ignition/gui/Plugin.hh"
+#include "ignition/gui/MainWindow.hh"
+#include <QTreeView>
+#include <QTreeWidget>
+#include "TopicViewer.hh"
+
+#define NAME_ROLE 51
+#define TYPE_ROLE 52
+#define TOPIC_ROLE 53
+#define PATH_ROLE 54
+#define PLOT_ROLE 55
+
+
+int g_argc = 1;
+char **g_argv = new char *[g_argc];
+
+using namespace ignition;
+using namespace gui;
+using namespace plugins;
+
+/////////////////////////////////////////////////
+TEST(TopicViewerTest, Load)
+{
+    common::Console::SetVerbosity(4);
+
+    Application app(g_argc, g_argv);
+    app.AddPluginPath(std::string(PROJECT_BINARY_PATH) + "/lib");
+
+    EXPECT_TRUE(app.LoadPlugin("TopicViewer"));
+
+    // Get main window
+    auto win = app.findChild<MainWindow *>();
+    ASSERT_NE(nullptr, win);
+
+    // Get plugin
+    auto plugins = win->findChildren<Plugin *>();
+    EXPECT_EQ(plugins.size(), 1);
+
+    auto plugin = plugins[0];
+    EXPECT_EQ(plugin->Title(), "Topic Viewer");
+
+    // Cleanup
+    plugins.clear();
+}
+
+TEST(TopicViewerTest, Model)
+{
+    // =========== Publish =================
+    transport::Node node;
+
+    // int
+    auto pubInt = node.Advertise<msgs::Int32>("/IntTopic");
```

Comment:
It's best practice to use `snake_case` for topic names, so I suggest:

```suggestion
    auto pubInt = node.Advertise<msgs::Int32>("/int_topic");
```